tabs/tab_logit.py

```
@@ -38,799 +35,402 @@ def check_perfect_separation(df, target_col):
-    # ========== SETTINGS ==========
-    with st.expander("⚙️ Advanced Settings", expanded=False):
-        col1, col2 = st.columns(2)
+    # --- 1. Dataset Selection Logic ---
+    @reactive.Calc
+    def current_df():
+        if is_matched.get() and input.radio_dataset_source() == "matched":
+            return df_matched.get()
+        return df.get()
+
+    @render.ui
+    def ui_dataset_selector():
```

Ruff (0.14.10)

Missing return type annotation for private function `ui_dataset_selector`

(ANN202)